dev/reference/best-practices/massive-regions.md

```diff
@@ -0,0 +1,143 @@
+
+Enable `Region Merge` by setting the following configurations:
+
+{{< copyable "" >}}
+
+```
+>> pd-ctl config set max-merge-region-size 20
+>> pd-ctl config set max-merge-region-keys 200000
+>> pd-ctl config set merge-schedule-limit 8
+```
+
+Refer to [Region Merge](https://github.com/tikv/tikv/blob/master/docs/how-to/configure/region-merge.md) and [PD configuration file](/dev/reference/configuration/pd-server/configuration-file.md#schedule) for more details.
+
+The default configuration of the `Region Merge` parameters is rather conservative. You can speed up the `Region Merge` process by referring to the method provided in [PD Scheduling Best Practices](/dev/reference/best-practices/pd-scheduling.md#region-merge-is-slow).
+
+## Other problems and solutions
+
+### Switching PD Leader is slow
+
+PD needs to persist Region Meta information on etcd to ensure that PD can quickly resume to provide Region routing services after switching the PD Leader node. As the number of Regions increases, the performance problem of etcd appears, making it slower for PD to get Region Meta information from etcd when PD is switching the Leader. With millions of Regions, it might take dozens or even tens of seconds to get the meta information from etcd.
+
+To address this problem, `use-region-storage` is enabled by default in PD since TiDB v3.0. With this feature enabled, PD stores Region Meta information on local LevelDB and synchronizes the information among PD nodes through other mechanism.
+
+### PD routing information is not updated in time
+
+In TiKV, pd-worker regularly reports Region Meta information to PD. When TiKV is restarted or switches the Region Leader, PD needs to recalculate Region's `approximate size / keys` through statistics. Therefore, with a large number of Regions, the single-threaded pd-worker might become the bottleneck, causing tasks to be piled up and not processed in time. In this situation, PD cannot obtain certain Region Meta information in time so that the routing information is not updated in time. This problem does not affect the actual reads and writes, but might cause inaccurate PD scheduling and require several round trips when TiDB updates Region cache.
+
+You can check *Worker pending tasks* under *Task* in the *TiKV Grafana* panel to determine whether pd-worker has tasks piled up. Generally, `pending tasks` should be kept at a relatively low value.
+
+![Check pd-worker](/media/best-practices/pd-worker-metrics.png)
+
+Currently, pd-worker is optimized for better efficiency in [#5620](https://github.com/tikv/tikv/pull/5620) on [TiKV master](https://github.com/tikv/tikv/tree/master), which is applied since [v3.0.5](/dev/releases/3.0.5.md#tikv). If you encounter a similar problem, it is recommended to upgrade to v3.0.5 or later version.
```

Comment:
```suggestion
Currently, pd-worker is optimized for better efficiency in [#5620](https://github.com/tikv/tikv/pull/5620) on [TiKV master](https://github.com/tikv/tikv/tree/master), which is applied since [v3.0.5](/dev/releases/3.0.5.md#tikv). If you encounter a similar problem, it is recommended to upgrade to v3.0.5 or later versions.
```